cmake/SSGCommon.cmake

```diff
@@ -761,6 +761,32 @@ macro(ssg_make_all_tables PRODUCT)
     )
 endmacro()
 
+macro(ssg_build_rule_dir_json)
+    add_custom_command(
+        OUTPUT "${CMAKE_BINARY_DIR}/rule_dirs.json"
+        COMMAND env "PYTHONPATH=$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/utils/rule_dir_json.py" --output "${CMAKE_BINARY_DIR}/rule_dirs.json" --root "${CMAKE_SOURCE_DIR}"
```

Comment:
@jan-cerny @matejak Thoughts on generating the `build/rule_dirs.json` by default?

I know there were discussions about this in the past, but couldn't find them.

Reply:
My concern would be that the `utils/rule_dir_json.py` is very slow, ~ 12 s on my beef laptop.

Reply:
The `rule_dirs.json` is used to collect the Ids of rules applicable to the product:
https://github.com/ComplianceAsCode/content/blob/28a5fb54a949065fbed57a9c6e39e5ddc931961a/utils/create_scap_delta_tailoring.py#L56

But as we are plugging the script into the build system, we could use the data that is available as part of the build pipeline. So collecting the Ids of rules applicable to the product can done by listing the files in `build/${PRODUCT}/rules`.

But I also see the value in being able to run `create_scap_delta_tailoring.py` without doing the whole build.
So the script could either receive a `rule_dir_json.py` or a `--resolved-rules-dir`?

During the build the target `generate-ssg-delta-${PRODUCT}-${PROFILE}` would depend on `${PRODUCT}-compile-all`.